website/src/apis/mpe.ts

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios';
+import getLocalStorage from 'storage/localStorage';
+import { Location, History } from 'history';
+import NUSModsApi from './nusmods';
+import { MpePreference } from '../types/mpe';
+import type { Module, ModuleCode } from '../types/modules';
+import { NUS_AUTH_TOKEN } from '../storage/keys';
+
+type MpePreferencesResponse = {
+  nusExchangeId: string;
+  preferences: Array<{
+    rank: number;
+    moduleCode: ModuleCode;
+    moduleType: MpePreference['moduleType'];
+  }>;
+};
+
+export class MpeSessionExpiredError extends Error {}
+
+const SSO_PATH = '/auth/sso';
+const MPE_PATH = '/mpe/submissions';
+const TOKEN_URL_QUERY = 'token';
+
+const storage = getLocalStorage();
+
+const getToken = (): string | null => storage.getItem(NUS_AUTH_TOKEN);
+
+const setToken = (token: string): void => storage.setItem(NUS_AUTH_TOKEN, token);
+
+const removeToken = (): void => storage.removeItem(NUS_AUTH_TOKEN);
+
+const mpe = axios.create({
+  baseURL: '/api/nus',
+});
+
+mpe.interceptors.request.use((config) => {
+  const updatedConfig = { ...config };
+  updatedConfig.headers.Authorization = getToken();
+  return updatedConfig;
```

Comment:
This does not immutably update `config` - `config.headers` is mutated. I'm not sure if that is important, but I'd suggest either this if immutability is not important 

```suggestion
  config.headers.Authorization = getToken();
```

or this if it is 

```suggestion
mpe.interceptors.request.use(produce(draft => {
  draft.headers.Authorization = getToken();
}));
```